zk/hermez_db/db_xlayer.go

```diff
@@ -59,3 +66,51 @@ func (db *HermezDbReader) GetInnerTxs(blockNum uint64) [][]*types.InnerTx {
 	}
 	return blockInnerTxs
 }
+
+// TruncateInnerTx deletes all inner txs of a block
+func (db *HermezDb) TruncateInnerTx(block uint64) error {
+	prefix := make([]byte, 8)
+	binary.BigEndian.PutUint64(prefix, block)
+
+	it, err := db.tx.Prefix(INNER_TX, prefix)
+	if err != nil {
+		log.Error("inner txs fetching failed", "err", err)
+		return nil
+	}
+	defer func() {
+		if casted, ok := it.(kv.Closer); ok {
+			casted.Close()
+		}
+	}()
+
+	var keyList [][]byte
+	for it.HasNext() {
+		k, v, err := it.Next()
+		if err != nil {
+			log.Error("inner txs fetching failed", "err", err)
+			return nil
+		}
+		innerTxs := make([]*types.InnerTx, 0)
+		if err := rlp.DecodeBytes(v, &innerTxs); err != nil {
+			err = fmt.Errorf("inner txs unmarshal failed:  %w", err)
+			log.Error("inner txs fetching failed", "err", err)
+			return nil
+		}
+		keyCopy := make([]byte, len(k))
+		copy(keyCopy, k)
+		keyList = append(keyList, keyCopy)
+	}
+
+	for _, k := range keyList {
+		err = db.tx.Delete(INNER_TX, k)
+		if err != nil {
+			log.Error("inner txs fetching failed", "err", err)
+			return err
+		}
+	}
+
+	afterTxs := db.GetInnerTxs(block)
+	afterCount := len(afterTxs)
+	log.Info("Delete inner txs", "block", block, "delete count", len(keyList), "after count", afterCount)
```

Comment:
Lose the count variable and reword english words for clarity 😅.
```suggestion
	log.Info("Delete inner txs", "block", block, "expected deleted count", len(keyList), "actual deleted count", len(afterTxs))
```